### PR TITLE
feat(support): add template info to support bundle

### PR DESCRIPTION
**Pull request by johnstcn**
@Emyrk raised that this would potentially be helpful for debugging certain types of workspace-related issues.

NOTE: there is currently an issue with how we generate ZIP files from TAR files. Fixing in a separate PR.

### CODE REVIEW

File: support/support_test.go [hunk 1]
```diff
@@ -136,10 +141,27 @@ func assertSanitizedDeploymentConfig(t *testing.T, dc *codersdk.DeploymentConfig
 }
 
 func setupWorkspaceAndAgent(ctx context.Context, t *testing.T, client *codersdk.Client, db database.Store, user codersdk.CreateFirstUserResponse) (codersdk.Workspace, codersdk.WorkspaceAgent) {
+	// This is a valid zip file
+	zipBytes := make([]byte, 22)
+	zipBytes[0] = 80
+	zipBytes[1] = 75
+	zipBytes[2] = 0o5
+	zipBytes[3] = 0o6
```
Comment (johnstcn) [inline support/support_test.go:149]: note: the formatter made these octal of its own volition

File: cli/support_test.go [hunk 2]
```diff
@@ -114,7 +114,6 @@ func assertBundleContents(t *testing.T, path string) {
 	require.NoError(t, err, "open zip file")
 	defer r.Close()
 	for _, f := range r.File {
-		require.NotZero(t, f.UncompressedSize64, "file %q should not be empty", f.Name)
```
Comment (johnstcn) [inline cli/support_test.go:117]: note: removing this as it's redundant with other checks

File: cli/support_test.go [hunk 3]
```diff
@@ -156,11 +155,26 @@ func assertBundleContents(t *testing.T, path string) {
...
+		case "workspace/template.json":
+			var v codersdk.Template
+			decodeJSONFromZip(t, f, &v)
+			require.NotEmpty(t, v, "workspace template should not be empty")
+		case "workspace/template_version.json":
+			var v codersdk.TemplateVersion
+			decodeJSONFromZip(t, f, &v)
+			require.NotEmpty(t, v, "workspace template version should not be empty")
+		case "workspace/parameters.json":
+			var v []codersdk.TemplateVersionParameter
```
Comment (mtojek) [inline cli/support_test.go:167]: `TemplateVersionParameter` or `WorkspaceBuildParameter`?

BTW There are also Template Version variables, but these may carry sensitive data.

Comment (johnstcn) [reply cli/support_test.go:167]: Ah, good catch. These are workspace build parameters. Will update 👍 

Agreed, we should leave out template version variables.

**Review by mtojek — APPROVED** [review]
👍

**Pull request merged**